blueocean-dashboard/src/main/js/components/IfCapability.jsx

```diff
@@ -0,0 +1,42 @@
+import React, { Component, PropTypes } from 'react';
+import {
+    actions,
+    capabilities as capabilitiesSelector,
+    createSelector,
+    connect,
+} from '../redux';
+const { object, array, func, string , bool} = PropTypes;
+
+export default class IfCapability extends Component {
+    constructor(props) {
+        super(props);
+        const { dataClass } = this.props;
+        this.props.fetchCapabilitiesIfNeeded(dataClass);
+
+    }
+    render() {
+        const { dataClass, capabilities ,capability } = this.props;
+
+        //Exit early, we havent fetched the caps yet.
+        if(!capabilities || !capabilities[dataClass]) {
+            return false;
+        }
+
+        //console.log("caps",caps);
+        if(capabilities[dataClass].find((cap => cap === capability))) {
+            return this.props.children;
+        } else {
+            return null;
+        }
+    }
+};
+
+IfCapability.propTypes = {
+    dataClass: string,
+    capability: string,
+};
+
+
+const selectors = createSelector([capabilitiesSelector], (capabilities) => ({ capabilities }));
```

Comment:
We have started to not connect certain component to redux on that level. It is more efficient to do this line and and the following in the PipelinePage and then in that page you need to switch to

```
{React.cloneElement(this.props.children, { pipeline,  capabilities})}
```

You have an early out already, so your component should still work the same, just get injected the capabilities from somewhere else.